backend/packages/Upgrade/src/api/services/CacheService.ts

```diff
@@ -17,25 +17,32 @@ export class CacheService {
   }
 
   public setCache<T>(id: string, value: T): Promise<T> {
-    return this.memoryCache.set(id, value);
+    return this.memoryCache ? this.memoryCache.set(id, value) : Promise.resolve(null);
   }
 
   public getCache<T>(id: string): Promise<T> {
-    return this.memoryCache.get(id);
+    return this.memoryCache ? this.memoryCache.get(id) : Promise.resolve(null);
   }
 
-  public delCache<T>(id: string): Promise<T> {
-    return this.memoryCache.del(id);
+  public delCache(id: string): Promise<void> {
+    return this.memoryCache ? this.memoryCache.del(id) : Promise.resolve();
   }
 
-  public resetCache(): void {
-    this.memoryCache.reset();
+  public async resetPrefixCache(prefix: string): Promise<void> {
+    const keys = this.memoryCache ? await this.memoryCache.store.keys() : [];
+    const filteredKeys = keys.filter((str) => str.startsWith(prefix));
```

Comment:
@RidhamShah Do we need this filteredKeys? It is just getting logged in next line. If not required, you can remove this and the console log as well.

Reply:
solved this